plugins/mcreator-core/datalists/blockstateproperties.yaml

```diff
@@ -0,0 +1,818 @@
+  type: Integer
+  other:
+    registry_name: stage
+    min: 0
+    max: 1
+
+- FACING:
+  readable_name: FACING
+  type: Enum
+  other:
+    registry_name: facing
+    values:
+      - down
+      - up
+      - north
+      - south
+      - west
+      - east
+- FACING_HOPPER:
+  readable_name: FACING_HOPPER
+  type: Enum
+  other:
+    registry_name: facing
+    values:
+      - down
+      - north
+      - south
+      - west
+      - east
+- HORIZONTAL_FACING:
+  readable_name: HORIZONTAL_FACING
+  type: Enum
+  other:
+    registry_name: facing
+    values:
+      - north
+      - south
+      - west
+      - east
+- VERTICAL_DIRECTION:
+  readable_name: VERTICAL_DIRECTION
+  type: Enum
+  other:
+    registry_name: facing
+    values:
+      - down
+      - up
+- AXIS:
+  readable_name: AXIS
+  type: Enum
+  other:
+    registry_name: axis
+    values:
+      - x
+      - y
+      - z
+- HORIZONTAL_AXIS:
+  readable_name: HORIZONTAL_AXIS
+  type: Enum
+  other:
+    registry_name: axis
+    values:
+      - x
+      - z
+
+- ATTACH_FACE:
+  readable_name: ATTACH_FACE
+  type: Enum
+  other:
+    registry_name: face
+    values:
+      - floor
+      - wall
+      - ceiling
+- BAMBOO_LEAVES:
+  readable_name: BAMBOO_LEAVES
+  type: Enum
+  other:
+    registry_name: leaves
+    values:
+      - none
+      - small
+      - large
+- BED_PART:
+  readable_name: BED_PART
+  type: Enum
+  other:
+    registry_name: part
+    values:
+      - head
+      - foot
+- BELL_ATTACHMENT:
+  readable_name: BELL_ATTACHMENT
+  type: Enum
+  other:
+    registry_name: attachment
+    values:
+      - floor
+      - ceiling
+      - single_wall
+      - double_wall
+- CHEST_TYPE:
+  readable_name: CHEST_TYPE
+  type: Enum
+  other:
+    registry_name: type
+    values:
+      - single
+      - left
+      - right
+- DOOR_HINGE:
+  readable_name: DOOR_HINGE
+  type: Enum
+  other:
+    registry_name: hinge
+    values:
+      - left
+      - right
+- DOUBLE_BLOCK_HALF:
+  readable_name: DOUBLE_BLOCK_HALF
+  type: Enum
+  other:
+    registry_name: half
+    values:
+      - upper
+      - lower
+- DRIPSTONE_THICKNESS:
+  readable_name: DRIPSTONE_THICKNESS
+  type: Enum
+  other:
+    registry_name: thickness
+    values:
+      - tip_merge
+      - tip
+      - frustum
+      - middle
+      - base
+- HALF:
+  readable_name: HALF
+  type: Enum
+  other:
+    registry_name: half
+    values:
+      - top
+      - bottom
+- MODE_COMPARATOR:
+  readable_name: MODE_COMPARATOR
+  type: Enum
+  other:
+    registry_name: mode
+    values:
+      - compare
+      - subtract
+- NOTEBLOCK_INSTRUMENT:
+  readable_name: NOTEBLOCK_INSTRUMENT
+  type: Enum
+  other:
+    registry_name: instrument
+    values:
+      - harp
+      - basedrum
+      - snare
+      - hat
+      - bass
+      - flute
+      - bell
+      - guitar
+      - chime
+      - xylophone
+      - iron_xylophone
+      - cow_bell
+      - didgeridoo
+      - bit
+      - banjo
+      - pling

```

Comment:
There are more in 1.21.1:

![image](https://github.com/user-attachments/assets/54331080-202c-49e3-8354-c65929325cc7)

However, this raises very concerning think I hoped we could avoid. How will we handle differences between MC versions, I hoped enums would remain more or less same.

This PR may become quite complex if we make mapping for every of those enum lists

Reply:
Note block instruments would also be needed by block itself when we convert materials to standalone settings... How about importing values from data list in cases when YAML find a string value of `values:` field instead of a list? This would also filter entries by whether they are supported in a specific version.

Reply:
This could work. You just need to figure out how to pass this information to templates so templates

Since this PR is almost completely reviewed, please make a new PR to this PR instead, so we can review this separately so we don't get back to square 0 here

Reply:
> This would also filter entries by whether they are supported in a specific version.

Filtering needs to be primarily done on generator/template level

Reply:
> You just need to figure out how to pass this information to templates

We can place tilde at the beginning of enum properties 3rd mapping (the class name).

Reply:
Make PR and then we will see how that looks and if it is needed

Reply:
Defeatomizer#114